### PR TITLE
Bring over all the Trainings configuration and tools

### PR DESCRIPTION
**Pull request by stevepiercy**
- Begin replacing "Trainings" with "Documentation"

See checklist in #1134

### CODE REVIEW

**Comment by stevepiercy** [comment]
This is a WIP. I still have a bit more work to do before it is ready for others to start using.

But if you just can't wait, check out this branch and read `docs/contributing/setup-build.md` for a quick start.

**Comment by stevepiercy** [comment]
This PR is now ready for review.

- Creates the structure shown in https://github.com/plone/documentation/issues/1137 Volto is still TBD.
- Add Makefile
- Add filter by section
- Use MyST and Sphinx Book Theme
- Add sticky TOC in navigation
- Copy code with a click
- Add Glossary
- Automatically generated index
- Add Contributing chapter

This gives us a good base from which to work.

**Comment by stevepiercy** [comment]
I just added GitHub actions for testing and (eventually) automatic deployment, which will come in https://github.com/plone/documentation/issues/1146

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+copybutton_prompt_is_regexp = True
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = "sphinx_book_theme"
+
```
Comment (ksuess) [inline docs/conf.py:167]: custom.css and print.css are not included in _build. I've build with `make html` and `gulp`.
With explicit html_static_path they are included.

```suggestion

html_static_path = ["_static"]

```

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+# See http://sphinx-doc.org/ext/todo.html#confval-todo_include_todos
+todo_include_todos = True
+
+# Announce that we have an opensearch plugin
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_use_opensearch
+html_use_opensearch = "https://docs.plone.org"
+
+html_theme_options = {
+    "google_analytics_id": "UA-1907133-6",
```
Comment (ksuess) [inline docs/conf.py]: Is the Google Analytics Tracking-Code for dev different from Tracking-Code of docs.plone.org?

**Review by ksuess — CHANGES_REQUESTED** [review]
Add robots.txt 
with html_extra_path https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_extra_path

**Comment by ksuess** [comment]
Ease to start contributing:

Maybe include creation of venv in `make build` :

```
python -m venv .
source bin/activate
```

and a hint to install enchant. Then make build would run without the need to read the guideline and preparing. And one bit of frustrating about the raised error is gone.

File: docs/conf.py (hunk 2 above)
Comment (stevepiercy) [reply docs/conf.py]: No, it is the same code.

Should I use a different one?

How do I get a different one?

Comment (ksuess) [reply docs/conf.py]: To not clutter the account of live documentation a different code would be good. Or maybe removing from dev and opening a reminding issue to bring it back on relaunch is sufficient.

File: docs/conf.py (hunk 1 above)
Comment (stevepiercy) [reply docs/conf.py:167]: Ah, yes, thank you! Fixed in de0b83a5b234f7909c6c2d9883003dcad0d1055b

**Comment by stevepiercy** [comment]
> Add robots.txt with html_extra_path https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_extra_path

Fixed in de0b83a5b234f7909c6c2d9883003dcad0d1055b

**Comment by stevepiercy** [comment]
> Ease to start contributing:
> 
> Maybe include creation of venv in `make build` :
> 
> ```
> python -m venv .
> source bin/activate
> ```

I moved around some things in the Makefile to get this to partially work.

Nothing I tried would activate the virtual environment and keep it activated for repeated `make html`. Instead I changed the value for `SPHINX_BUILD` to use its relative path.

> and a hint to install enchant. Then make build would run without the need to read the guideline and preparing. And one bit of frustrating about the raised error is gone.

This is in the `contributing/setup-build.md` file. Maybe we do an `if/fi` to check if enchant is in the path, else exit?

**Comment by stevepiercy** [comment]
Nope, that breaks the build on Netlify. I reverted the Makefile change and updated the instructions in `setup-build.md` in bb6c179bfff8b289251ef1d94872d7a0a4a3a94d.

File: docs/conf.py (hunk 2 above)
Comment (stevepiercy) [reply docs/conf.py]: Done in 95404a22c142a3e7a5dc36c34e10755583f7df89 and https://github.com/plone/documentation/issues/1158.

**Comment by stevepiercy** [comment]
@ksuess I've pushed some commits and responded to your review. Tag! You're it! :grin:

I was not able to figure out the virtual environment activation. I also don't think we need to do much more in the Makefile for enchant, as the instructions is the entry point for getting started.

**Comment by ksuess** [comment]
`make build` works quite fine now. It's not even necessary to activate the venv. There's some magic that sphinx-build is found. Try 
`$ sphinx-build `
and
`$ which sphinx-build`

So you can skip the line
`source bin/activate`
in setup-build.md

**Comment by ksuess** [comment]
> `make build` works quite fine now. It's not even necessary to activate the venv. There's some magic that sphinx-build is found. Try `$ sphinx-build ` and `$ which sphinx-build`
> 
> So you can skip the line `source bin/activate` in setup-build.md

It's wrong what I wrote, I did a test from scratch. But I think it's fine now. We can come back to it later.

**Comment by stevepiercy** [comment]
@ksuess, I added one more commit for you to test out. You'll need to run `pip install -r requirements` then `make livehtml`.

If it is OK to merge this PR as is, please let me know. I am eager to get something deployed for people to start playing with.

**Comment by ksuess** [comment]
This can be deleted:


.PHONY: test
testlight: clean spellcheck  ## Run spellcheck

**Comment by stevepiercy** [comment]
> This can be deleted:
> 
> .PHONY: test testlight: clean spellcheck ## Run spellcheck

Good catch! Fixed in 694129236df6e52e441e2fa5903d735114e9bdcc.

**Comment by ksuess** [comment]
`make livehtml` 😻

**Comment by ksuess** [comment]
Maybe just add a text hint with `@echo`:

.PHONY: build
build:  ## Set up training: Install requirements
	python3 -m venv . || virtualenv --clear --python=python3 .
	bin/python -m pip install --upgrade pip
	bin/pip install -r requirements.txt
	@echo
	@echo "Please activate your Python virtual environment with"
	@echo "source bin/activate"

**Comment by ksuess** [comment]
Let's see what other reviewers find. From me a GO!

**Comment by ksuess** [comment]
> 



> `make livehtml` 😻

I you add 
```
.PHONY: livehtml
livehtml: ## Run live updating build
```
then `make livehtml `is shown in the output  of `make`

**Comment by stevepiercy** [comment]
I added updates per your comments, but grabbed the description from https://pypi.org/project/sphinx-autobuild/ for `make`. See 4d035adcd7389b76972b24b880c1c75ffb84dbda

**Comment by ksuess** [comment]
As long as this PR is open: here another little one:
_source folder with txt files could be excluded via robots.txt. Maybe more than this folder…
image
For example https://61e8de5e50d8f50008c486f8--6-dev-docs-plone-org.netlify.app/_sources/glossary.md.txt

**Comment by stevepiercy** [comment]
I updated the robots.txt, based on what shows up in `_build/html/`.

**Comment by stevepiercy** [comment]
I'm going to merge this branch after Europeans have had one more day to review it.

I want to announce that `6-dev` is ready for everyone to work on before the weekend.

File: .github/workflows/build_deploy.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        run: make linkcheck
+
+#      - name: Prepare deploy
+#        run: make deploy
+#
+#      - name: Deploy to server
+#        id: deploy
+#        uses: Pendect/action-rsyncer@v1.1.0
+#        env:
+#          DEPLOY_KEY: ${{secrets.DEPLOY_KEY_TRAINING}}
```
Comment (polyester) [inline .github/workflows/build_deploy.yml:38]: This probably needs a new key

File: .github/workflows/build_deploy.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#        with:
+#          flags: '-avzr --delete'
+#          options: ''
+#          ssh_options: '-p ${{secrets.DEPLOY_PORT}}'
+#          src: '_build/html/'
+#          dest: '${{secrets.DEPLOY_USER_TRAINING}}@${{secrets.DEPLOY_SERVER_TRAINING}}:${{secrets.DEPLOY_PATH_TRAINING}}/5'
```
Comment (polyester) [inline .github/workflows/build_deploy.yml:44]: here also

File: .github/workflows/build_deploy.yml (hunk 3 above)
Comment (stevepiercy) [reply .github/workflows/build_deploy.yml:38]: Yup, that's why I commented it out. Who can provide one?

Comment (polyester) [reply .github/workflows/build_deploy.yml:38]: @ericof should be able to do so.

Comment (stevepiercy) [reply .github/workflows/build_deploy.yml:38]: Let's continue the conversion in https://github.com/plone/documentation/issues/1146

**Pull request merged**